### PR TITLE
#6987 add theme step in context creator

### PR DESCRIPTION
**Pull request by MV88**
## Description

A new step is added to the context creation for the theme selection
the list of themes is overridable with **themes** property in ContextCreator plugin

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#6987

**What is the new behavior?**

the theme will be saved here in the local state when loaded
image

and directly in contextcreator.selectedTheme during customization

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
image

### CODE REVIEW

**Comment by tdipisa** [comment]
@MV88 can you provide also a screenshot?

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/contextcreator/ConfigureThemes.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        <div className="configure-themes-step">
+            <div className="choose-theme">
+                <ControlLabel><Message msgId="contextCreator.configureThemes.themes"/></ControlLabel>
+                <Select
+                    onChange={({value})  => setSelectedTheme(value)}
+                    value={selectedTheme}
+                    options={themes}
+                    disabled={false}
+                    noResultsText="contextCreator.configureThemes.noThemes"
+                />
```
Comment (allyoucanmap) [inline web/client/components/contextcreator/ConfigureThemes.jsx:30]: I did not test this but should the change needed for the previous comment
```suggestion
                <Select
                    onChange={(option)  => setSelectedTheme(option?.theme)}
                    value={selectedTheme}
                    options={themes.map(theme => ({ value: theme.id, label: theme.label || theme.id, theme }))}
                    disabled={false}
                    noResultsText="contextCreator.configureThemes.noThemes"
                />
```

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/contextcreator/ContextCreator.jsx [hunk 2]
```diff
@@ -162,6 +166,7 @@ export default class ContextCreator extends React.Component {
         contextNameChecked: true,
         newContext: {},
         resource: {},
+        themes: [{value: "default", label: "Default"}, {value: "dark", label: "Dark"}],
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/components/contextcreator/ContextCreator.jsx:169]: sure

File: web/client/components/contextcreator/ConfigureThemes.jsx (hunk 1 above)
Comment (MV88) [reply web/client/components/contextcreator/ConfigureThemes.jsx:30]: i think we need to use option.id

**Comment by MV88** [comment]
sorry i messed up with commits. closed in favor of https://github.com/geosolutions-it/MapStore2/pull/7043

**Pull request closed without merging**